c_glib/arrow-glib/compute.cpp

```diff
@@ -1448,6 +1448,35 @@ garrow_array_is_in(GArrowArray *left,
   }
 }
 
+/**
+ * garrow_array_argsort:
+ * @array: A #GArrowArray.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Returns: (nullable) (transfer full): The indices that would sort
+ *   an array on success, %NULL on error.
+ *
+ * Since: 0.15.0
+ */
+GArrowArray *
+garrow_array_argsort(GArrowArray *array,
```

Comment:
How about using other name than "argsort" because "argsort" isn't consistent with our existing API?

  * We use "_" as word separator in function name.
  * We don't use abbreviated name as much as possible.

"argsort" is familiar to NumPy users because NumPy uses "argsort": https://docs.scipy.org/doc/numpy/reference/generated/numpy.argsort.html

But "argsort" isn't widely name outside of NumPy.

For example, R uses "order": https://cran.r-project.org/doc/manuals/r-release/fullrefman.pdf#Rfn.order

MATLAB's "sort" returns sorted array and indices to sort array: https://mathworks.com/help/matlab/ref/sort.html

Here are candidates of this function name from me:

  1. `order`: Like R
  2. `sort`: Like MATLAB but doesn't return sorted array itself.
  3. `sort_index`: Add suffix for `sort` variable. This is our API naming rule that adds suffix for variant.
  4. `non_materialized_sort`: We used "non-materializing sort" in https://issues.apache.org/jira/browse/ARROW-1566 . It well describes but it's a bit long.
      * Or `sort_non_materialized`

In my opinion: 4 -> 3 -> 1 -> 2

@shiro615 @mrkn What do you think about this?

Reply:
I think `non_materialized_sort` is better.
I think `sort_index` is also good, but pandas's `sort_index`(`pandas.DataFrame.sort_index`) returns the sorted data frame itself.

Reply:
One more candidate:

  5. `sort(options)`
      * `options::materialize`: `gboolean`. Only `FALSE` is valid for now.

I like 5 -> 4 -> 3 -> 1 -> 2.

Reply:
I suspect that `argsort` came from `argmax`. Julia also has argmax, but doesn’t have argsort. Instead, it has [sortperm](https://pkg.julialang.org/docs/julia/THl1k/1.1.1/base/sort.html#Base.sortperm) for the feature of numpy’s argsort.

I'm unconfident that `non_materialized_sort` is good for this feature because of the following two reasons. (1) `non_materialized_sort` is too long.  (2) I'm not sure that "non-materialized" intuitively means "making a sorted index array".  The meaning of "materialize" is similar to "appear".  So "non-materialized sort" tells us it doesn't make a sorted value array, but it doesn't say that it makes a sorted index array.

I propose that the new candidate: `sort(index: false)`.  This is short and intuitive.

Reply:
The disadvantage of switching return value by argument is that the type of a return value is different by the argument.

Providing both `sort_values` and `sort_indicies` doesn’t introduce the ambiguity of the return value.

Reply:
It's generally yes. But this is not occurred for this case because we have generic `GArrowArray` class. `sort_values()` returns `GArrowArray *` and `sort_indices()` returns `GArrowUInt64Array *`. But we can use `GArrowArray *` for `sort_indices()` too.

If Arrow C++ accepts `SortIndices`, we'll use `GArrowUInt64Array *garrow_array_sort_indices()`.